src/main/java/cn/nukkit/entity/Entity.java

```diff
@@ -1234,8 +1234,12 @@ public boolean entityBaseTick(int tickDiff) {
             if (this.fireTicks <= 0) {
                 this.extinguish();
             } else if (!this.fireProof) {
-                this.setDataFlag(DATA_FLAGS, DATA_FLAG_ONFIRE, true);
-                hasUpdate = true;
+                if (this instanceof Player && ((Player) this).isSpectator()) {
+                    // do nothing
```

Comment:
If nothing is going in this if statement block then the conditions can be inverted and also added to the `!this.fireProof` statement.